sechub-wrapper-xray/src/main/java/com/mercedesbenz/sechub/wrapper/xray/cli/XrayWrapperArtifactoryClientController.java

```diff
@@ -0,0 +1,119 @@
+package com.mercedesbenz.sechub.wrapper.xray.cli;
+
+import java.util.Objects;
+import java.util.concurrent.TimeUnit;
+
+import org.cyclonedx.model.Bom;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.mercedesbenz.sechub.wrapper.xray.api.XrayAPIArtifactoryClient;
+import com.mercedesbenz.sechub.wrapper.xray.config.XrayWrapperArtifact;
+import com.mercedesbenz.sechub.wrapper.xray.config.XrayWrapperConfiguration;
+import com.mercedesbenz.sechub.wrapper.xray.report.XrayWrapperReportException;
+import com.mercedesbenz.sechub.wrapper.xray.report.XrayWrapperReportReader;
+
+public class XrayWrapperArtifactoryClientController {
+
+    private static final Logger LOG = LoggerFactory.getLogger(XrayWrapperArtifactoryClientController.class);
+    private final XrayWrapperConfiguration xrayWrapperConfiguration;
+    private final XrayAPIArtifactoryClient artifactoryClient;
+    private int retries;
```

Comment:
retries is stateful here. As long as `waitForScansToFinishAndDownloadReport` is called only one time it is not problematic. But the wait method is public and if somebody would reuse the controller instance, the second call would no work any more (I expect that a new wait would start from 0.